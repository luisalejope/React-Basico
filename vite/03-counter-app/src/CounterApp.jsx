import { useState } from 'react'
import PropTypes from 'prop-types'

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value)

    const handleAdd = () => setCounter(counter + 1)
    

    const handleSubstract = () => {
        if (counter > 0) {
            setCounter(counter - 1)
        }
    }

    const handleReset = () => setCounter(value)

    return (
        <>
            <h1>CounterAPP</h1>
            <h2>{counter}</h2>
            <button aria-label='btn-add' onClick={handleAdd}>+1</button>
            <button aria-label='btn-reset' onClick={handleReset}>reset</button>
            <button aria-label='btn-rest' onClick={handleSubstract}>-1</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}