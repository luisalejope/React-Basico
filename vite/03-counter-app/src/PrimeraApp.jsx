import PropTypes from 'prop-types';
// functional component

export const PrimeraApp = ({ saludo, subtitulo }) => {
  /* const persona = {
    nombre: "Luis",
    apellido: "Peña",
    edad: "45",
    clave: "ironman",
    direccion: {
      ciudad: "Bogotá",
      zip: "1222231",
      lat: "14.53242",
      long: "32.131",
    },
  }; */

  return (
    <>
      <h1 data-testid='test-saludo'>{saludo}</h1>
      {/* <pre> {JSON.stringify(persona, null, 3)} </pre> */}
      <p>{subtitulo}</p>
    </>
  );
};
// Obliga los argumentos y el tipo
PrimeraApp.propTypes ={
    saludo: PropTypes.string.isRequired,
    subtitulo: PropTypes.string
}
// Define las propiedades por defecto
PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
}
