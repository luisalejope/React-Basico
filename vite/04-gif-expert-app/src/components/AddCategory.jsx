import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({ onAddCategory }) => {

    const [inputValue, setInputValue] = useState('')
    const onInputChange = (e) => setInputValue(e.target.value)
    

    const onSubmit = (e) => {
        e.preventDefault()

        if (inputValue.trim().length <= 1) {
            return;
        }

        onAddCategory(inputValue.trim())
        setInputValue('')
    }


  return (
    <form onSubmit={onSubmit} aria-label="form">  
        <input type="text" placeholder="Buscar Gifs" onChange={onInputChange} value={inputValue}/>
    </form>
  )
}


AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired
}

