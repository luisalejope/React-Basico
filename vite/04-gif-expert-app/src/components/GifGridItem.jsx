import PropTypes from 'prop-types'

export const GifGridItem = ({title, url, id}) => {
  return (
    <div className='card' id={id}>
        <h2>
            {title.trim() !== '' ? title : 'GIF SIN TITULO'}
        </h2>
        <img src={url} alt={title} />
    </div>
  )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    id: PropTypes.string
}
