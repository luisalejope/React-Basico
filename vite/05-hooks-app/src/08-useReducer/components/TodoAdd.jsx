import PropTypes from 'prop-types'
import { useForm } from '../../hooks/useForm'

export const TodoAdd = ({ handleSubmit }) => {
    const { todoInput, onInputChange, onResetForm } = useForm({
        todoInput: '',
    })

    const onAddTodo = (e) => {
        e.preventDefault()

        if (todoInput === '') return;

        handleSubmit(todoInput)
        onResetForm();
    }

    return (
        <form>
            <input type="text" className="form-control" name='todoInput' onChange={onInputChange} value={todoInput} />
            <button type="submit" className="btn btn-primary mt-2" onClick={onAddTodo}>Agregar</button>
        </form>
    )
}

TodoAdd.propTypes = {
    handleSubmit: PropTypes.func.isRequired
}