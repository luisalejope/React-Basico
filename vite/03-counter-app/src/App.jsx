// import { CounterApp } from './CounterApp'
import {PrimeraApp} from './PrimeraApp'

export const App = () => {
  return (
    <>
      {/* <CounterApp value={1} /> */}
      <PrimeraApp saludo='Hola'/>
    </>
  )
}
