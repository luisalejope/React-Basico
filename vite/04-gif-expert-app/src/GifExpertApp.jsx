import { useState } from "react"
import { AddCategory, GifGrid } from "./components"

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch'])

    const onAddCategory = (newCategory) => {

        const alreadyExists = categories.includes(newCategory)

        if (!alreadyExists) {
            setCategories([newCategory, ...categories])
        }
    }

    return (
        <>
            <div>GifExpertApp</div>

            <AddCategory onAddCategory={onAddCategory} />

            <ol key={'hola'} className="categories">
                {categories.map(category => (
                    <GifGrid category={category} key={category} />
                ))}
            </ol>
        </>
    )
}
