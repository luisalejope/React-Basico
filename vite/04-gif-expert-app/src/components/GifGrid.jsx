import PropTypes from 'prop-types'
import { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'


export const GifGrid = ({ category }) => {

    const {gifs, isLoading} = useFetchGifs(category)
    
    return (
        <>
            <h3>{category}</h3>
            <div className='card-grid'>
                { isLoading ? <p>Loading...</p> : gifs.map( gif => (
                    <GifGridItem key={gif.id} {...gif} />
                ))}
            </div>
        </>
    )
}
/* 4hJUgNoRRsiqFKOAr3qnnLRzwwkh0dZM */
GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
