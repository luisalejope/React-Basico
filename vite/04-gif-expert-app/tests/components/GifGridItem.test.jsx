import { render, screen } from "@testing-library/react"
import { GifGridItem } from "../../src/components"

describe('Tests on <GifGidItem />', () => {
    // 1. Inicializar
    const title = 'Titulo de un gif'
    const url = 'https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=5a057a20fpp96z4mez6lqnh2014rf965nl0ex79ivce4m420&ep=v1_gifs_search&rid=giphy.gif&ct=g'

    test('should match with the snapshot', () => {

        // 2. Etimulo
        const {container} = render(<GifGridItem title={title} url={url} />)

        //3. Observar el comportamiento
        expect(container).toMatchSnapshot()
    })


    test('should show the image wwith the URL and the ALT sended', () => {
        // 2. Estimulo
        render(<GifGridItem title={title} url={url} />)
        const {src, alt} = screen.getByRole('img')
        // 3. Observar el comportamiento
        expect(alt).toBe(title)
        expect(src).toBe(url);
    })

    test('should show the title with the info sended', () => {
        // 2. Estimulo
        render(<GifGridItem title={title} url={url} />)
        const titleRendered = screen.getByRole('heading', {level: 2}).innerHTML
        // 3. Observar el comportamiento
        expect(titleRendered).toBe(title)
    })

})