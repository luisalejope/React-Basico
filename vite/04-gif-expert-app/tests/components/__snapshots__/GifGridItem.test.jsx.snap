// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGidItem /> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <h2>
      Titulo de un gif
    </h2>
    <img
      alt="Titulo de un gif"
      src="https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=5a057a20fpp96z4mez6lqnh2014rf965nl0ex79ivce4m420&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
  </div>
</div>
`;
